tools/c7n_azure/c7n_azure/resources/event_hub.py

```diff
@@ -46,3 +53,38 @@ class resource_type(ArmResourceManager.resource_type):
             'properties.isAutoInflateEnabled'
         )
         resource_type = 'Microsoft.EventHub/namespaces'
+
+
+@EventHub.filter_registry.register('firewall-rules')
+class EventHubFirewallRulesFilter(FirewallRulesFilter):
+
+    def __init__(self, data, manager=None):
+        super(EventHubFirewallRulesFilter, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.eventhub')
+        self.client = None
+
+    @property
+    def log(self):
+        return self._log
+
+    def process(self, resources, event=None):
+        self.client = self.manager.get_client()
+        return super(EventHubFirewallRulesFilter, self).process(resources, event)
+
+    def _query_rules(self, resource):
+        query = self.client.namespaces.get_network_rule_set(
+            resource['resourceGroup'],
+            resource['name'])
+
+        resource_rules = IPSet()
```

Comment:
I don't think we need the for loop that happens next for handling CIDR blocks and single addresses separately, I think this will work:

`resource_rules = IPSet([r.ip_mask for r in query.ip_rules])`

example at: 
https://repl.it/repls/OptimisticLemonchiffonScreencast